prov/usnic/src/usdf_msg.c

```diff
@@ -827,8 +827,8 @@ usdf_msg_handle_recv(struct usdf_domain *udp, struct usd_completion *comp)
 
 		rx_ptr = (uint8_t *)(pkt + 1);
 		rxlen = ntohs(pkt->msg.m.rc_data.length);
+		rqe_ptr = (uint8_t *)(rqe->ms_cur_ptr + rqe->ms_length);
 		rqe->ms_length += rxlen;
-		rqe_ptr = (uint8_t *)rqe->ms_cur_ptr;
```

Comment:
I'm not sure this change is right in the case of a `recvv`.  If the user posts an iovec with `recvv` that looks like this:
```c
{ 
    [0] = {
        .iov_base = (void *)0x1000;
        .iov_len = 4;
    },
    [1] = {
        .iov_base = (void *)0xf000000;
        .iov_len = 40000;
    }
}
```

I don't see anything that will ever update `rqe->ms_cur_ptr` to point into the buffer in the second iov entry.  I think the initial assignment of `rqe_ptr = (uint8_t *)rqe->ms_cur_ptr` is correct but we need to make sure that at the end of the loop that copies from `rxbuf` we need to update `rqe->ms_cur_ptr` to point to the right point.  I think that's just assigning `rqe->ms_cur_ptr = rqe_ptr`, but I could be missing something.